src/api/custom-resolvers/mutations/create-an-application/index.ts

```diff
@@ -0,0 +1,121 @@
+import getAccountById from '../../../helpers/get-account-by-id';
+import getCountryByField from '../../../helpers/get-country-by-field';
+import { CreateAnApplicationVariables, Context } from '../../../types';
+
+/**
+ * createAnApplication
+ * Create an application.
+ * 1) Get a country from a provided isoCode.
+ * 2) Create a new application with owner relationship.
+ * 3) Create eligibility relationship with the country and application.
+ * 4) Create buyer relationship with the country and application.
+ * 5) Update the application with buyer and eligibility IDs.
+ * @param {Object} GraphQL root variables
+ * @param {Object} GraphQL variables for the CreateAnApplication mutation
+ * @param {Object} KeystoneJS context API
+ * @returns {Object} Object with success flag
+ */
+const createAnApplication = async (root: any, variables: CreateAnApplicationVariables, context: Context) => {
+  console.info('Creating application for ', variables.accountId);
+
+  try {
+    const { accountId, eligibilityAnswers } = variables;
+
+    /**
+     * Check the account exists.
+     * If not, reject.
+     */
+    const account = await getAccountById(context, accountId);
+
+    if (!account) {
+      return {
+        success: false,
+      };
+    }
+
+    /**
+     * Get a country's ID from the provided ISO code.
+     * This is required to be used in:
+     * 1) Eligibility buyer country relationship
+     * 2) Buyer country relationship
+     */
+    const { buyerCountryIsoCode, ...otherEligibilityAnswers } = eligibilityAnswers;
+
+    const country = await getCountryByField(context, 'isoCode', buyerCountryIsoCode);
+
+    /**
+     * Create the initial application.
+     * We need to do this first so that we can use the application ID.
+     */
+    const application = await context.db.Application.createOne({
+      data: {
+        owner: {
+          connect: { id: accountId },
+        },
+      },
+    });
+
+    /**
+     * Create eligibility and relationships for:
+     * 1) The buyer country
+     * 2) The application
+     */
+    const eligibility = await context.db.Eligibility.createOne({
```

Comment:
this file is quite long.  What if we put these in their own files? like eligibility and buyer

Reply:
Great shout, i've moved eligibility and buyer into helper functions. 

Would be great to do the same with `Application`, but there are few instances of this, would like to do this (along with other context methods), in another PR as it would make this PR even bigger. Will create a ticket if that's ok.